### ISSUE TITLE
cabal list-bin is broken

### ISSUE THREAD

**Issue by athas** [haskell/cabal#8664]
Since b2be49cad5adc43d82dc2199199f298f455f51f1 (#8622) `cabal list-bin` no longer produces any output on any project I try it on.  If I change the use of `info` back to `putStrLn`, it works again.

Steps to reproduce the behavior: In the cabal repository itself: `cabal list-bin cabal`

It is also possible that I am holding it wrong (I don't understand this entire `info` machinery), but I don't see how a command like `list-bin` can possibly require any complexity or subtle behaviour.

**Comment by athas** [comment]
Changing the flag passed to `info` to `verbose` also makes it work, but perhaps someone savvy about this system should weigh in.  The `info` function does nothing if the passed-in verbosity isn't at least `verbose`, so to me it makes no sense to ever pass it a constant.

**Comment by ulysses4ever** [comment]
@athas great catch, thanks! I think, the right way to do it is `notice verbosity` instead of the current `info normal`. Could you try that and, if satisfied, submit a patch, please?

**Comment by fgaz** [comment]
I think we shouldn't use logging functions for that. This isn't a log, it's normal output. I'd expect `list-bin` to print out the path even with `-v0`.

**Comment by fgaz** [comment]
And we really need a test for list-bin. This isn't the first time it breaks

**Comment by ulysses4ever** [comment]
@fgaz that makes sense. If you read the initial ticket by @andreasabel #8400, it seems there was an argument for that had to do with the desire to have some more control than what what a mere `putStrLn` gives. But perhaps, it's not what we want and indeed, your example with `-v0` makes perfect sense.

**Comment by andreasabel** [comment]
The intention of #8400 was to make it possible to get a test for `list-bin` in our current test framework (which catches marked output).  In #8622, no test was added; it was a mistake to approve it nevertheless, my apologies.
Maybe the immediate action would be to revert #8622, and reopen #8400, and then fix #8400 properly _with_ test.

Refs:
- #8400 
- #8622

**Comment by athas** [comment]
That sounds like right way to go for now.

**Comment by athas** [comment]
And I suppose an actual solution might just use `withOutputMarker` directly, but someone familiar with the Verbosity system should probably weigh in.

**Comment by andreasabel** [comment]
Turns out we had a test case for `list-bin` contributed in #7925, but it didn't catch the main output (path to executable) because of #8400.  A correct fix of #8400 should have touched the [output of this testcase](https://github.com/haskell/cabal/blob/master/cabal-testsuite/PackageTests/ListBin/Script/cabal.out).
There is even [another test](https://github.com/haskell/cabal/blob/bcfc79ce1e286b1df1fef31139044c0e0503d5c7/cabal-testsuite/PackageTests/ListBin/SelectedComponent/cabal.test.hs) contributed by @bacchanalia which should also benefit from a proper fix of #8400.  I feel like both of these tests have to be revised to actually check that the path to the binary is produced by `list-bin` (atm, they succeed without checking that).

Ref:
- https://github.com/haskell/cabal/tree/bcfc79ce1e286b1df1fef31139044c0e0503d5c7/cabal-testsuite/PackageTests/ListBin

**Comment by andreasabel** [comment]
I have an emergency fix at #8668 (revert #8622) and a proper fix at #8670 which makes the emergency fix obsolete if merged quickly enough.

**Comment by ulysses4ever** [comment]
I also screwed up with approving the patch. Thank you Andreas for quickly acting on this!

**Comment by Bodigrim** [comment]
Reopening, the fix in #8670 misses a newline character. Depending on your terminal emulator it might look a bit different. 

In `zsh`:
```
$ cabal-3.8.1.0 list-bin bytestring-tests
/Users/andrew/programs/bytestring/dist-newstyle/build/aarch64-osx/ghc-9.2.5/bytestring-0.11.3.1/t/bytestring-tests/build/bytestring-tests/bytestring-tests
```
but look at the last character:
```
$ cabal list-bin bytestring-tests
Warning: this is a debug build of cabal-install with assertions enabled.
/Users/andrew/programs/bytestring/dist-newstyle/build/aarch64-osx/ghc-9.2.5/bytestring-0.11.3.1/t/bytestring-tests/build/bytestring-tests/bytestring-tests%
```

**Comment by Mikolaj** [comment]
We'd also need a new test that checks the newline.

**Comment by andreasabel** [comment]
> We'd also need a new test that checks the newline.

This isn't directly in the scope of the test framework, since it adds the newline: https://github.com/haskell/cabal/blob/a78d77b7edd758b9cda3617111445397ac4e2673/Cabal/src/Distribution/Simple/Utils.hs#L622-L625

**Comment by andreasabel** [comment]
> Reopening, the fix in #8670 misses a newline character.

Since I am assigned to this issue, I will suggest a PR.
